pkg/config/model/types.go

```diff
@@ -103,6 +103,9 @@ type ReaderWriter interface {
 type Setup interface {
 	// API implemented by viper.Viper
 
+	// SetReady should be called when Setup is done and the config is ready to be used
+	SetReady()
```

Comment:
WDYT about renaming this to `BuildConfig` or `Generate` ? `SetReady` doesn't clearly imply that we will load all files and env vars once called.

Reply:
`BuildConfig` stutters, which is less than ideal, but I think it's okay for now. One thing is, right now the `datadog.yaml` file gets read a bit later (the [config component here](https://github.com/DataDog/datadog-agent/blob/66de8311bec89875e3b80aa7a1383bb9cc1158ae/comp/core/config/config.go#L91) calls `setupConfig`, [which calls](https://github.com/DataDog/datadog-agent/blob/66de8311bec89875e3b80aa7a1383bb9cc1158ae/comp/core/config/setup.go#L58) `LoadWithSecret`), so [the place where](https://github.com/DataDog/datadog-agent/pull/30361/files#diff-a7b6619528f155e64cefe12cc4ba753eb7688c87fdf9d96e1052a12eeb519df5R266) `BuildConfig` gets called is only constructing 2 of the sources (default and env). Maybe a name that implies it's only building the "schema" instead?